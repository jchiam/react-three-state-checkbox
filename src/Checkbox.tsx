import * as React from 'react';

export interface CheckboxProps extends React.Props<Checkbox> {
  checked: boolean;
  indeterminate?: boolean;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
  onChange?: (e: React.FormEvent) => void;
}

export default class Checkbox extends React.Component<CheckboxProps, any> {
  static defaultProps: Partial<CheckboxProps> = {
    indeterminate: false,
    className: '',
    style: {},
    disabled: false
  };

  render() {
    const { className, style, checked, indeterminate, onChange, disabled } = this.props;
    return (
      <input
        type="checkbox"
        className={className}
        style={style}
        ref={input => {
          /* istanbul ignore else */
          if (input) {
            input.checked = checked;
            input.indeterminate = indeterminate as boolean;
            input.disabled = disabled as boolean;
          }
        }}
        onChange={(e: React.FormEvent) => {
          /* istanbul ignore else */
          if (onChange) {
            onChange(e);
          }
        }}
      />
    );
  }
}
