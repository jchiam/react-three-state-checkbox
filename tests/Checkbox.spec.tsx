import * as React from 'react';
import { configure, shallow, mount } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

import Checkbox from '../src/Checkbox';

describe('<Checkbox />', () => {
  it('renders checkbox', () => {
    const tree = mount(<Checkbox checked />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders checkbox with the right default props', () => {
    const tree = mount(<Checkbox checked={false} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders checkbox with all props defined', () => {
    const tree = mount(<Checkbox checked indeterminate className="test-class" style={{ margin: 0 }} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
