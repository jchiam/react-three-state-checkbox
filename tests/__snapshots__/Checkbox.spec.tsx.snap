// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders checkbox 1`] = `
<Checkbox
  checked={true}
  className=""
  disabled={false}
  indeterminate={false}
  style={Object {}}
>
  <input
    className=""
    onChange={[Function]}
    style={Object {}}
    type="checkbox"
  />
</Checkbox>
`;

exports[`<Checkbox /> renders checkbox with all props defined 1`] = `
<Checkbox
  checked={true}
  className="test-class"
  disabled={false}
  indeterminate={true}
  onChange={[MockFunction]}
  style={
    Object {
      "margin": 0,
    }
  }
>
  <input
    className="test-class"
    onChange={[Function]}
    style={
      Object {
        "margin": 0,
      }
    }
    type="checkbox"
  />
</Checkbox>
`;

exports[`<Checkbox /> renders checkbox with the right default props 1`] = `
<Checkbox
  checked={false}
  className=""
  disabled={false}
  indeterminate={false}
  style={Object {}}
>
  <input
    className=""
    onChange={[Function]}
    style={Object {}}
    type="checkbox"
  />
</Checkbox>
`;
